src/ucp/core/ucp_mm.c

```diff
@@ -401,6 +409,13 @@ ucs_status_t ucp_mem_map(ucp_context_h context, const ucp_mem_map_params_t *para
             status = UCS_ERR_INVALID_PARAM;
             goto out;
         }
+
+        if (!((memory_type == UCS_MEMORY_TYPE_HOST) ||
+              (memory_type == UCS_MEMORY_TYPE_UNKNOWN))) {
+            ucs_error("memory allocation not supported with non-host memory");
+            status = UCS_ERR_INVALID_PARAM;
+            goto out;
+        }
```

Comment:
maybe check it in ucp_mem_map_common() where it can't be UCS_MEMORY_TYPE_UNKNOWN?
line 297